TeslaMateAgile.Tests/Services/PGEServiceTests.cs

```diff
@@ -0,0 +1,57 @@
+using Microsoft.Extensions.Options;
+using Moq;
+using Moq.Contrib.HttpClient;
+using NUnit.Framework;
+using TeslaMateAgile.Data.Options;
+using TeslaMateAgile.Services;
+
+namespace TeslaMateAgile.Tests.Services;
+
+public class PGEServiceTests
+{
+    private PGEService _subject;
+    private Mock<HttpMessageHandler> _handler;
+
+    [SetUp]
+    public void Setup()
+    {
+        _handler = new Mock<HttpMessageHandler>();
+        var httpClient = _handler.CreateClient();
+        var pgeOptions = Options.Create(new PGEOptions 
+        { 
+            BaseUrl = "https://pge-pe-api.gridx.com",
+            Utility = "PGE",
+            Market = "DAM",
+            RateName = "EV2A",
+            RepresentativeCircuitId = "083611114",
+            Program = "CalFUSE"
+        });
+        httpClient.BaseAddress = new Uri(pgeOptions.Value.BaseUrl);
+        _subject = new PGEService(httpClient, pgeOptions);
```

Comment:
There are test failures around here, see the test run - make sure the new tests build and pass correctly (and doesn't break any existing ones!)